core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -448,6 +448,15 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
         }
       }
     }
+
+    // System property spark.yarn.app.id must be set if user code ran by AM on a YARN cluster
+    // yarn-standalone is deprecated, but still supported
+    if ((get("spark.master") == "yarn-cluster" || get("spark.master") == "yarn-standalone") &&
```

Comment:
What we do here is actually an act of validating the configuration. It seems to be a good place inside `validateSettings`. There are no information required from outside to do this check.